astropy/io/fits/hdu/nonstandard.py

```diff
@@ -52,7 +52,10 @@ def fromfile(cls, filename, compress=False):
             Gzip compress the FITS file
         """
 
-        return cls.fromhdulist(HDUList.fromfile(filename), compress=compress)
+        hdulist = HDUList.fromfile(filename)
```

Comment:
Just curious, why not use the context manager here?

Reply:
Does `fromfile` have a context manager form? I have ever only used context manager with `open`.

Reply:
I didn't think about it, thanks @MSeifert04.
@pllim , yes the context manager is on `HDUList`, and `fits.open` is just a shortcut for `HDUList.fromfile`.